cluster-autoscaler/cloudprovider/aws/aws_manager.go

```diff
@@ -295,14 +296,18 @@ func extractTaintsFromAsg(tags []*autoscaling.TagDescription) []apiv1.Taint {
 	for _, tag := range tags {
 		k := *tag.Key
 		v := *tag.Value
-		splits := strings.Split(k, "k8s.io/cluster-autoscaler/node-template/taint/")
-		if len(splits) > 1 {
-			values := strings.SplitN(v, ":", 2)
-			taints = append(taints, apiv1.Taint{
-				Key:    splits[1],
-				Value:  values[0],
-				Effect: apiv1.TaintEffect(values[1]),
-			})
+		// The tag value must be in the format <tag>:NoSchedule
+		r, _ := regexp.Compile("(.*):NoSchedule")
```

Comment:
This seems overly specific, don't we care about taints with other effects? I assume the out of bounds exceptions happened when getting values[1], how about checking that len(values) > 1

Reply:
You're right, I forgot about the other taints.